File: Printpress.UI/Printpress_Angular/src/app/features/order/models/order/order-upsert.Dto.ts
import { IObjectState } from "../../../../core/models/i-object-state";
import { OrderServicesUpsertDTO } from "../order-service/order-service-upsertDto";
import { OrderGroupUpsertDto } from "../orderGroup/order-group-upsert.Dto";
import { OrderSellingItemUpsertDto } from "../orderSellingItem/order-selling-item-upsert.dto";

export class OrderUpsertDto extends IObjectState {

    constructor(
        public id: string,
        public name: string,
        public clientId: string,
        public orderGroups: OrderGroupUpsertDto[],
        public orderServices: OrderServicesUpsertDTO[],
        public sellingItems: OrderSellingItemUpsertDto[]) {

        super();
    }
}

File: Printpress.Application/Orders/DTO/Order/Upsert/OrderUpsertDto.cs
using System.ComponentModel.DataAnnotations;

namespace Printpress.Application;

public class OrderUpsertDto : TrackedDTO
{
    public Guid Id { get; set; }

    [Required]
    public string Name { get; set; }

    [Required]
    public Guid ClientId { get; set; }

    public List<OrderGroupUpsertDTO> OrderGroups { get; set; }
    public List<OrderServiceUpsertDTO> OrderServices { get; set; }
    public List<OrderSellingItemUpsertDTO> SellingItems { get; set; }
}
